src/InputFilter.php

```diff
@@ -1016,23 +1016,62 @@ private function cleanHtml($source)
 	 */
 	private function cleanPath($source)
 	{
-		$linuxPattern = '/^[A-Za-z0-9_\/-]+[A-Za-z0-9_\.-]*([\\\\\/]+[A-Za-z0-9_-]+[A-Za-z0-9_\.-]*)*$/';
+		// Linux and other Unixoids
+		$filePattern          = '(?:[^\x00\/:*?]{1,255})';
+		$pathSeparatorPattern = '(?:\/+)';
+		$rootPattern          = '(?:\/)';
 
-		if (preg_match($linuxPattern, $source))
+		if ($this->pathMatches($source, $rootPattern, $pathSeparatorPattern, $filePattern, '/'))
 		{
-			return preg_replace('~/+~', '/', $source);
+			return $source;
 		}
 
-		$windowsPattern = '/^([A-Za-z]:(\\\\|\/))?[A-Za-z0-9_-]+[A-Za-z0-9_\.-]*((\\\\|\/)+[A-Za-z0-9_-]+[A-Za-z0-9_\.-]*)*$/';
+		// Windows
+		$filePattern          = '(?:[^\x00\\\\\/:*"?<>|]{1,255})';
+		$pathSeparatorPattern = '(?:[\\\\\/])';
+		$rootPattern          = '(?:[A-Za-z]:(\\\\|\/))';
 
-		if (preg_match($windowsPattern, $source))
+		if ($this->pathMatches($source, $rootPattern, $pathSeparatorPattern, $filePattern, '\\'))
 		{
-			return preg_replace('~(\\\\|\/)+~', '\\', $source);
+			return $source;
 		}
 
 		return '';
 	}
 
+	/**
+	 * Fix a path, if and only if it matches the provided patterns.
+	 *
+	 * If a path matches but is longer than 4095 bytes, it is cleared.
+	 *
+	 * @param   string  $source                The path as provided; it gets cleaned in place, if possible.
+	 * @param   string  $rootPattern           The pattern to identify an absolute path (e.g., '/' on Linux, 'C:\' on Windows),
+	 * @param   string  $pathSeparatorPattern  The pattern for valid path separators
+	 * @param   string  $filePattern           The pattern for valid file and directory names
+	 * @param   string  $pathSeparator         The native path separator
+	 *
+	 * @return bool
```

Comment:
```suggestion
	 * @return boolean
```
Fix PHPCS.

Reply:
In addition, we have many deprecated messages due to curly bracket syntax used for arrays, see https://ci.joomla.org/joomla-framework/filter/63/1/3 .

Reply:
... But those seem not to be related to this PR, or I get the log wrong.

Reply:
Fixed